app/containers/RegisterEmail/actions.js

```diff
@@ -98,7 +107,7 @@ export function register() {
 
     const role_id = role === 'attendee' ? 1 : role === 'booth' ? 2 : 3;
 
-    if (first_name && role && email && password && username) {
+    if (first_name && role && email && password && username && ValidateEmail(email)) {
```

Comment:
you could nest one of the checking of `validateEmail()` and the other fields, so you can show the alert in a much better way.